### ISSUE TITLE
Installation issue: OpenMPI@5.0.{1,2} incorrect pmix dependency version range

### ISSUE THREAD

**Issue by amd-toolchain-support** [LLNL/spack#42651]
### Steps to reproduce the issue

Maintainers: @hppritcha  @naughtont3 

OpenMPI versions 5.0.x require `pmix@4.2.4:` as a dependency when not using the bundled pmix.  The expressed dependency range in the recipe however is `pmix@4.2:`.  This leads to issues with e.g `pmix@4.2.2` being reused from OpenMPI@4.x installations by the concretizer, which then fails at build time with the below error.

```console
$ spack install openmpi@4.1.4
...
$ spack spec openmpi@5.0.2
Input spec
--------------------------------
 -   openmpi@5.0.2

Concretized
--------------------------------
 -   openmpi@5.0.2%gcc@11.2.1~atomics~cuda~gpfs~internal-hwloc~internal-libevent~internal-pmix~java~legacylaunchers~lustre~memchecker~openshmem~orterunprefix~romio+rsh~static+vt+wrapper-rpath build_system=autotools fabrics=none schedulers=none arch=linux-rocky9-x86_64_v3
[+]      ^autoconf@2.72%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^m4@1.4.19%gcc@11.2.1+sigsegv build_system=autotools patches=9dc5fbd,bfdffa7 arch=linux-rocky9-x86_64_v3
[+]              ^diffutils@3.9%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^libsigsegv@2.14%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]      ^automake@1.16.5%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]      ^gcc-runtime@11.2.1%gcc@11.2.1 build_system=generic arch=linux-rocky9-x86_64_v3
[+]      ^gmake@4.4.1%gcc@11.2.1~guile build_system=generic arch=linux-rocky9-x86_64_v3
[+]      ^hwloc@2.9.1%gcc@11.2.1~cairo~cuda~gl~libudev+libxml2~netloc~nvml~oneapi-level-zero~opencl+pci~rocm build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v3
[+]          ^libpciaccess@0.17%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^util-macros@1.19.3%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^libxml2@2.10.3%gcc@11.2.1+pic~python+shared build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^libiconv@1.17%gcc@11.2.1 build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v3
[+]              ^xz@5.4.6%gcc@11.2.1~pic build_system=autotools libs=shared,static arch=linux-rocky9-x86_64_v3
[+]          ^ncurses@6.4%gcc@11.2.1~symlinks+termlib abi=none build_system=autotools arch=linux-rocky9-x86_64_v3
[+]      ^libevent@2.1.12%gcc@11.2.1+openssl build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^openssl@3.2.1%gcc@11.2.1~docs+shared build_system=generic certs=mozilla arch=linux-rocky9-x86_64_v3
[+]              ^ca-certificates-mozilla@2023-05-30%gcc@11.2.1 build_system=generic arch=linux-rocky9-x86_64_v3
[+]      ^libtool@2.4.7%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^findutils@4.9.0%gcc@11.2.1 build_system=autotools patches=440b954 arch=linux-rocky9-x86_64_v3
[+]      ^numactl@2.0.14%gcc@11.2.1 build_system=autotools patches=4e1d78c,62fc8a8,ff37630 arch=linux-rocky9-x86_64_v3
[+]      ^openssh@9.5p1%gcc@11.2.1+gssapi build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^krb5@1.20.1%gcc@11.2.1+shared build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^bison@3.8.2%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^gettext@0.22.4%gcc@11.2.1+bzip2+curses+git~libunistring+libxml2+pic+shared+tar+xz build_system=autotools arch=linux-rocky9-x86_64_v3
[+]                  ^tar@1.34%gcc@11.2.1 build_system=autotools zip=pigz arch=linux-rocky9-x86_64_v3
[+]                      ^pigz@2.8%gcc@11.2.1 build_system=makefile arch=linux-rocky9-x86_64_v3
[+]                      ^zstd@1.5.5%gcc@11.2.1+programs build_system=makefile compression=none libs=shared,static arch=linux-rocky9-x86_64_v3
[+]          ^libedit@3.1-20210216%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]          ^libxcrypt@4.4.35%gcc@11.2.1~obsolete_api build_system=autotools patches=4885da3 arch=linux-rocky9-x86_64_v3
[+]      ^perl@5.38.0%gcc@11.2.1+cpanm+opcode+open+shared+threads build_system=generic patches=714e4d1 arch=linux-rocky9-x86_64_v3
[+]          ^berkeley-db@18.1.40%gcc@11.2.1+cxx~docs+stl build_system=autotools patches=26090f4,b231fcc arch=linux-rocky9-x86_64_v3
[+]          ^bzip2@1.0.8%gcc@11.2.1~debug~pic+shared build_system=generic arch=linux-rocky9-x86_64_v3
[+]          ^gdbm@1.23%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]              ^readline@8.2%gcc@11.2.1 build_system=autotools patches=bbf97f1 arch=linux-rocky9-x86_64_v3
[+]      ^pkgconf@1.9.5%gcc@11.2.1 build_system=autotools arch=linux-rocky9-x86_64_v3
[+]      ^pmix@4.2.2%gcc@11.2.1~docs+pmi_backwards_compatibility~python~restful build_system=autotools arch=linux-rocky9-x86_64_v3
[+]      ^zlib-ng@2.1.5%gcc@11.2.1+compat+opt build_system=autotools arch=linux-rocky9-x86_64_v3
```

### Error message

<details><summary>Error message</summary>

<pre>
    --- SNIP ---
     2516    checking for pmix pkg-config libs... -lpmix
     2517    checking for pmix pkg-config static libs... -lpmix -ldl -levent_co
             re -levent_pthreads -lhwloc -lm -lxml2 -lpthread -llzma -liconv -l
             pciaccess -lhwloc -lm -lpthread -lz -lxml2 -lz -llzma -liconv -lm 
             -lpciaccess
     2518    checking for pmix.h... yes
     2519    checking for PMIx_Init... yes
     2520    checking version at or above v4.2.4... no
     2521    configure: WARNING: PRRTE requires PMIx v0x00040204 or above.
  >> 2522    configure: error: Please select a supported version and configure 
             again
     2523    configure: ===== done with 3rd-party/prrte configure =====
  >> 2524    configure: error: PRRTE configuration failed.  Cannot continue.
  --- SNIP ---
</pre></details>

### Information on your system

spack debug report
* **Spack:** 0.22.0.dev0 (4a04989bbb633b2c53deaead7a177be57aca11c5)
* **Python:** 3.9.10
* **Platform:** linux-rocky9-zen4
* **Concretizer:** clingo

### Additional information

[spack-build-env.txt](https://github.com/spack/spack/files/14261943/spack-build-env.txt)
[spack-build-out.txt](https://github.com/spack/spack/files/14261944/spack-build-out.txt)

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [X] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [X] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by hppritcha** [comment]
i'll see about reproducing this.

**Comment by psakievich** [comment]
Has this issue been resolved by #42827?

**Comment by hppritcha** [comment]
yes it should be resolved now via #42827
